### PR TITLE
layers: More error message to use Location

### PR DESCRIPTION
**Pull request by spencer-lunarg**
More error messages I found not using `Location`

some examples of improvements commented below

### CODE REVIEW

File: layers/core_checks/cc_render_pass.cpp [hunk 1]
```diff
@@ -4381,14 +4385,15 @@ bool CoreChecks::PreCallValidateCreateFramebuffer(VkDevice device, const VkFrame
                                     framebuffer_attachments_create_info->pAttachmentImageInfos[attachment_index].layerCount;
                                 if (layer_count <= highest_view_bit) {
                                     skip |= LogError(
-                                        pCreateInfo->renderPass, "VUID-VkFramebufferCreateInfo-renderPass-03198",
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_render_pass.cpp:4384]: > Validation Error: [ VUID-VkFramebufferCreateInfo-renderPass-03198 ] Object 0: handle = 0xee647e0000000009, type = VK_OBJECT_TYPE_RENDER_PASS; | MessageID = 0x7b11c6de | vkCreateFramebuffer(): pCreateInfo->pNext\<VkFramebufferAttachmentsCreateInfo\>.pAttachmentImageInfos[0].layerCount is 1, but the view mask for subpass 0 in renderPass (VkRenderPass 0xee647e0000000009[]) includes layer 1, with that attachment specified as a depth/stencil attachment.

File: layers/core_checks/cc_image.cpp [hunk 2]
```diff
@@ -864,28 +864,37 @@ static inline bool ContainsRect(VkRect2D rect, VkRect2D sub_rect) {
...
+
 bool CoreChecks::ValidateClearAttachmentExtent(const CMD_BUFFER_STATE &cb_state, const VkRect2D &render_area,
                                                uint32_t render_pass_layer_count, uint32_t rect_count,
-                                               const VkClearRect *clear_rects) const {
+                                               const VkClearRect *clear_rects, const Location &loc) const {
     bool skip = false;
 
     for (uint32_t i = 0; i < rect_count; i++) {
         if (!ContainsRect(render_area, clear_rects[i].rect)) {
-            skip |= LogError(cb_state.Handle(), "VUID-vkCmdClearAttachments-pRects-00016",
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_image.cpp:874]: > Validation Error: [ VUID-vkCmdClearAttachments-pRects-00016 ] Object 0: handle = 0x55e491869520, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x99777dbe | vkCmdClearAttachments(): pRects[0].rect (extent.width: 260, extent.height: 128, offset.x: 0, offset.y: 0) is not contained in the area of the current render pass instance (extent.width: 256, extent.height: 256, offset.x: 0, offset.y: 0).

File: layers/core_checks/cc_image.cpp [hunk 3]
```diff
@@ -1431,47 +1442,40 @@ bool CoreChecks::ValidateImageViewFormatFeatures(const IMAGE_STATE &image_state,
...
-        skip |= LogError(image_state.image(), "VUID-VkImageViewCreateInfo-usage-02275",
-                         "vkCreateImageView(): pCreateInfo->format %s with tiling %s does not support usage that includes "
-                         "VK_IMAGE_USAGE_STORAGE_BIT.",
-                         string_VkFormat(view_format), string_VkImageTiling(image_tiling));
+        skip |= LogError("VUID-VkImageViewCreateInfo-usage-02275", image_state.image(), create_info_loc.dot(Field::format),
+                         "%s with tiling %s only supports %s.", string_VkFormat(view_format), string_VkImageTiling(image_tiling),
+                         string_VkFormatFeatureFlags2(tiling_features).c_str());
     } else if ((image_usage & VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT) &&
                !(tiling_features & (VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT | VK_FORMAT_FEATURE_2_LINEAR_COLOR_ATTACHMENT_BIT_NV))) {
-        skip |= LogError(image_state.image(), "VUID-VkImageViewCreateInfo-usage-08931",
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_image.cpp:1450]: >  Validation Error: [ VUID-VkImageViewCreateInfo-usage-08931 ] Object 0: handle = 0xe88693000000000c, type = VK_OBJECT_TYPE_IMAGE; | MessageID = 0xca93720b | vkCreateImageView(): pCreateInfo->format VK_FORMAT_R32G32B32A32_SINT with tiling VK_IMAGE_TILING_OPTIMAL only supports VK_FORMAT_FEATURE_2_DEPTH_STENCIL_ATTACHMENT_BIT. The Vulkan spec states: If usage contains VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT, then the image view's format features must contain VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT or VK_FORMAT_FEATURE_2_LINEAR_COLOR_ATTACHMENT_BIT_NV

before we just repeated the spec

File: layers/core_checks/cc_image.cpp [hunk 4]
```diff
@@ -864,28 +864,37 @@ static inline bool ContainsRect(VkRect2D rect, VkRect2D sub_rect) {
     return true;
 }
 
+// TODO - Make a common util and find others places that would use this
+static std::string string_VkRect2D(VkRect2D rect) {
+    std::stringstream ss;
+    ss << "extent.width: " << rect.extent.width << ", extent.height: " << rect.extent.height << ", offset.x: " << rect.offset.x
```
Comment (artem-lunarg) [inline layers/core_checks/cc_image.cpp:870]: nit: at first offset then dimensions, I feel this order is more common

**Review by artem-lunarg — APPROVED** [review]

File: layers/core_checks/cc_image.cpp (hunk 4 above)
Comment (spencer-lunarg) [reply layers/core_checks/cc_image.cpp:870]: agree, while at it, going to look for other `VkRect2D`

**Pull request merged**